adapters/rubicon/rubicon.go

```diff
@@ -190,11 +190,31 @@ var rubiSizeMap = map[rubiSize]int{
 	rubiSize{w: 200, h: 600}:   126,
 }
 
-func lookupSize(s openrtb.Format) (int, error) {
-	if sz, ok := rubiSizeMap[rubiSize{w: uint16(s.W), h: uint16(s.H)}]; ok {
-		return sz, nil
+//MAS algorithm
+func findPrimary(alt []int) (int, []int) {
+	min, pos, primary := 0, 0, 0
+	for i, size := range alt {
+		if size == 15 {
+			primary = 15
+			pos = i
+			break
```

Comment:
why does this one break while size 9 and 2 don't?

Reply:
this is for selection of primary ad size. When we encounter a 15, we take 15 for primary and break. But for 9 and 2, there is still a chance that 15 might occur much later in the array so we don't break immediately.